### ISSUE TITLE
Feature: Support for modern image filtering algorithms.

### ISSUE THREAD

**Issue by infradragon** [jurplel/qView#646]
While bilinear filtering is fast, it is one of the least quality image filtering algorithms you could use. It matters more than you might expect, especially when trying to read text. I recommend you implement modern lanczos-type filtering algorithms like the lohalo and nohalo algorithms in gimp. 
It will make reading text in images much much easier and it will also significantly reduce ugly moire patterns (see [here](https://www.gimpusers.com/system/tutorial_steps/3070/Calliphora_methods-original.jpg)). 
Another useful option would be to use one interpolation method when downscaling, and another when upscaling, such as using nohalo or bilinear to downscale and using no interpolation to upscale, so you can still zoom in and look at the pixels.

Demonstration:
Original image
![BNLEP3ImageNew](https://github.com/jurplel/qView/assets/72665124/64309117-8301-499e-a503-8e04394a6346)

Downscaled, no interpolation (1/2 resolution)
![none](https://github.com/jurplel/qView/assets/72665124/7f004c9b-f123-440a-9e95-07213d97b210)

Downscaled, bilinear interpolation (1/2 resolution)
![bilinear](https://github.com/jurplel/qView/assets/72665124/d6a1de61-c7dc-4132-aa9c-132ca6ce05e1)

Downscaled, nohalo interpolation (1/2 resolution)
![nohalo](https://github.com/jurplel/qView/assets/72665124/5faa9da5-105f-4e06-ac24-208f6e087432)

Upscaled, no interpolation (4x resolution)
![none-up](https://github.com/jurplel/qView/assets/72665124/ca5187fd-b282-4f7c-b88f-f10e84cb1064)

Upscaled, bilinear interpolation (4x resolution)
![bilinear-up](https://github.com/jurplel/qView/assets/72665124/fdda7ea6-560e-40f9-bf0a-ec17b719c467)

Upscaled, nohalo interpolation (4x resolution)
![nohalo-up](https://github.com/jurplel/qView/assets/72665124/45959b0e-9476-4e62-9e9a-b9523cb64bff)

**Comment by infradragon** [comment]
lanczos is certainly the sharpest but it basically applies a sharpening filter after upscaling which is the reason i dont like it much. you can see sharpening artifacts on the lanczos upscale, especially on the cables and switch on the left. IMO the nohalo algorithm from gimp would be the best to use.

**Comment by infradragon** [comment]
another thing to note is that mitchell is also called "triangle" in some programs

**Comment by jurplel** [comment]
Yep, agree this would be a nice feature. bilinear is used because it is built into Qt and like you said, it is quite fast. Slower algorithms might require more carefulness with regard to how frequent the image is logically resized, but that may be a fine tradeoff.
